__tests__/integration.test.js

```diff
@@ -458,3 +458,145 @@ describe("DELETE /api/comments/:comment_id", () => {
       });
   });
 });
+
+describe("GET /api/articles (queries)", () => {
+  it("should  accept a topic query, serves up aricles by topic", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(12);
+        articles.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              topic: "mitch",
+            })
+          );
+        });
+      });
+  });
+  it("should serve up all articles if topic is ommited", () => {
+    return request(app)
+      .get("/api/articles?topic=")
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles.length).toBe(13);
+      });
+  });
+  it("should 404 for non existent topic", () => {
+    return request(app)
+      .get("/api/articles?topic=the flying peshwaris")
+      .expect("Content-Type", /json/)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("not found");
+      });
+  });
+  it("should 404 for bad datatype", () => {
+    return request(app)
```

Comment:
Is a 404 an appropriate status code for a bad datatype? If this is a bad topic search this should be a 400.